src/main/java/org/folio/services/budget/BudgetService.java

```diff
@@ -64,6 +68,17 @@ private SharedBudget mergeBudgets(SharedBudget sharedBudget, Budget budgetFromSt
       .withTotalFunding(budgetFromStorage.getAllocated() + budgetFromStorage.getNetTransfers());
   }
 
+  private CompletableFuture<Void> rollbackBudgetPutIfNeeded(Budget budgetFromStorage, Throwable t1, RequestContext requestContext) {
+    if (t1 == null) {
+      return CompletableFuture.completedFuture(null);
+    }
+    budgetFromStorage.setVersion(budgetFromStorage.getVersion() + 1);
```

Comment:
We should avoid setting versions manually, previsly we had issues with it and @grabsefx refactored all places to delete this

Reply:
Here is this story : https://issues.folio.org/browse/MODFISTO-340

Reply:
The story is not clear about how to fix this. I want to do a rollback to a previous version. If I simply use PUT with an older version number, it will fail. I tried using null for the version, but that didn't work. Should I use a GET and replace all the fields except _version with the old values before the PUT ?

Reply:
Here is corresponding PR to delete version usage in code: https://github.com/folio-org/mod-finance-storage/pull/284/files
We observed this issue when running python rollover script https://issues.folio.org/browse/MODFISTO-350
Also it was reported by another customers, so as a workaround we provided script that disables all OL triggers.
Usage of GET before PUT can be an option, @grabsefx what do you think?

Reply:
I think we should follow the RMB documentation regarding OL:
` The client should never increment the _version property; only the trigger should increment it to ensure that the check and the increment are in the same transaction to avoid any race condition`
So, yes, every time you want to update the record you should retrieve the latest version of it and modify all required fields except `_version`

Reply:
I started implementing that, but then I realized it was logically equivalent to setting the version, and much more complex. I will add a simple commit to get the latest version before the PUT so that I don't need to increment the version number and also to reduce conflict risks.